tests/test_entity.py

```diff
@@ -1,121 +1,259 @@
+    val[1] = 1
+    val[2] = 2
+    assert np.allclose(e_1.value, 42)
+    assert np.allclose(e_2.value, 42)
     val = np.ones((42, 42, 42, 3))
-    e = me.H(val)
+    e = me.Entity("ExternalMagneticField", value=val)
     assert np.allclose(e.value, val)
 
 
-def test_list_as_value():
-    val = [42, 42, 42]
-    e = me.Ku(val)
-    assert np.allclose(e.value, np.array(val))
+# %% initialize with quantity
+
+
+def test_init_quantity():
+    q = 1 * u.A / u.m
+    e = me.Entity("ExternalMagneticField", value=q)
+    assert u.allclose(e.quantity, q)
+    assert np.allclose(e.value, 1)
+    assert e.unit == u.A / u.m
+    q = 1 * u.kA / u.m
+    e = me.Entity("ExternalMagneticField", value=q, unit="kA/m")
+    assert u.allclose(e.quantity, q)
+    assert np.allclose(e.value, 1)
+    assert e.unit == u.kA / u.m
+    e = me.Entity("ExternalMagneticField", value=q)
+    assert u.allclose(e.quantity, q)
+    assert np.allclose(e.value, 1)
+    assert e.unit == u.kA / u.m
+    e = me.Entity("ExternalMagneticField", value=q, unit=" MA/m")
+    assert u.allclose(e.quantity, q)
+    assert np.allclose(e.value, 1e-3)
+    assert e.unit == u.MA / u.m
+
+
+# %% initialize with entity
+
+
+def test_init_entity():
+    e_1 = me.Entity("ExternalMagneticField", value=1)
+    e_2 = me.Entity("ExternalMagneticField", value=e_1)
+    assert u.allclose(e_1.quantity, e_2.quantity)
+    assert np.allclose(e_1.value, e_2.value)
+    assert e_1.unit == e_2.unit
+    e_3 = me.Entity("ExternalMagneticField", value=e_1, unit="mA/m")
+    assert u.allclose(e_3.quantity, e_1.quantity)
+    assert np.allclose(e_3.value, 1e3)
+    assert e_3.unit == u.mA / u.m
+
+
+def test_check_init_unit():
+    # change unit (conversion/change unit after initialized entity)
+    e = me.Entity("SpontaneousMagnetization", value=1, unit=u.A / u.m)
+    e.quantity.to("kA/m")
+    assert e.unit == u.A / u.m
+    e.quantity.to("kA/m", copy=False)
+    assert e.unit == u.A / u.m
+    with pytest.raises(u.UnitConversionError):
+        me.Entity("SpontaneousMagnetization", value=1, unit="T")
+    with (
+        u.set_enabled_equivalencies(u.magnetic_flux_field()),
+        pytest.raises(u.UnitConversionError),
+    ):
+        me.Entity("SpontaneousMagnetization", value=1 * u.T, unit="A/m")
 
 
-def test_tuple_as_value():
-    val = (42, 42, 42)
-    e = me.Ms(val)
-    assert np.allclose(e.value, np.array(val))
+# %% check attributes
 
 
-def test_entity_drop_ontology_numpy(onto_class_list):
-    for label in onto_class_list:
-        e = me.Entity(label, 42)
-        root_e = np.sqrt(e)
-        with pytest.raises(AttributeError):
-            _ = root_e.ontology
+def test_attrs_H():
+    e = me.Entity("ExternalMagneticField")
+    assert hasattr(e, "ontology_label")
+    assert hasattr(e, "ontology_label_with_iri")
+    assert hasattr(e, "ontology")
+    assert hasattr(e, "quantity")
+    assert hasattr(e, "value")
+    assert hasattr(e, "unit")
+    assert hasattr(e, "axis_label")
```

Comment:
I would drop this and instead in other test check that all attributes are defined *and* produce sensible output. We have done this extensively for `quantity`, `value` and `unit`. We should add checks for `ontology_label` in some of the previous tests. The rest I would test in dedicated tests. I assume a lot we will have in the tests to come.